draft-bbs-signatures.md

```diff
@@ -386,50 +386,66 @@ Procedure:
 
 This operation computes a deterministic signature from a secret key (SK) and optionally over a header and or a vector of messages.
 
+**Messages and generators input:**
+
+The generators and messages are supplied as a list of pairs (called `to_sign`) of the form `(H_i, msg_i)`. The generator `H_i` must be a point on G1 and the message `msg_i` must be a non-zero scalar mod r representing a message to be signed (see [Messages and Generators](#messages-and-generators)). The order of this list will determine the order in which the generators and messages should be provided to the [Verify](#verify), [ProofGen](#proofgen) and [ProofVerify](#proofverify) functions. This document does not specify how this order should be enforced, when communicating the signed messages between the participating parties. Implementations may elect to sort the pairs alphabetically or communicate them alongside indexes, indicating their order.
+
 ```
-signature = Sign(SK, PK, header, (msg_1,..., msg_L), (H_1,..., H_L))
+signature = Sign(SK, PK, header, to_sign)
 
 Inputs:
 
-- SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
-- PK (REQUIRED), an octet string of the form outputted by the SkToPk operation provided the above SK as input.
-- header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
-- msg_1,..., msg_L (OPTIONAL), a vector of octet strings.
-- H_1,..., H_L (OPTIONAL), points of G1. Generators used to commit each message.
+- SK (REQUIRED), an octet string of the form outputted by the KeyGen
+                 operation.
+- PK (REQUIRED), an octet string of the form outputted by the SkToPk
+                 operation provided the above SK as input.
+- header (OPTIONAL), an octet string containing context and application
+                     specific information. If not supplied, it defaults
+                     to an empty string.
+- to_sign (OPTIONAL), a list of (generator, message) pairs where the
+                      generator is a point in G1 and the message is a
+                      non-zero scalar mod r. If not supplied, will
+                      default to the empty array ("[]").
 
 Parameters:
 
-- Ciphersuite_ID (REQUIRED), ASCII string. The unique ID of the ciphersuite.
-- H_s (REQUIRED), point of G1. The generator for the blinding value of the signature.
-- H_d (REQUIRED), point of G1. The generator used to sign the signature domain.
+- Ciphersuite_ID (REQUIRED), ASCII string. The unique ciphersuite ID.
```

Comment:
Can remove the notion of (REQUIRED) here as all parameters are required

Reply:
I'm happy to make the change if people thing that to not be that big of an issue.